### PR TITLE
Update getPaginatedChartList to use namespace.

### PR DESCRIPTION
**Pull request by absoludity**
This PR cannot land without the related change to the handlers nor updating the frontend to always send the namespace (following PR).

Updates getPaginatedChartList to expect namespace with a value or "_all" and filter accordingly. Adds actual tests for postgres, but not mongodb.

Ref #1521

### CODE REVIEW

File: cmd/assetsvc/main.go [hunk 1]
```diff
@@ -42,22 +42,23 @@ func setupRoutes() http.Handler {
...
-	apiv1.Methods("GET").Path("/charts/{repo}/search").Queries("q", "{query}").Handler(WithParams(searchCharts))
-	apiv1.Methods("GET").Path("/charts/{repo}/search").Queries("q", "{query}", "showDuplicates", "{showDuplicates}").Handler(WithParams(searchCharts))
-	apiv1.Methods("GET").Path("/charts/{repo}/{chartName}").Handler(WithParams(getChart))
-	apiv1.Methods("GET").Path("/charts/{repo}/{chartName}/versions").Handler(WithParams(listChartVersions))
-	apiv1.Methods("GET").Path("/charts/{repo}/{chartName}/versions/{version}").Handler(WithParams(getChartVersion))
-	apiv1.Methods("GET").Path("/assets/{repo}/{chartName}/logo").Handler(WithParams(getChartIcon))
-	apiv1.Methods("GET").Path("/assets/{repo}/{chartName}/versions/{version}/README.md").Handler(WithParams(getChartVersionReadme))
-	apiv1.Methods("GET").Path("/assets/{repo}/{chartName}/versions/{version}/values.yaml").Handler(WithParams(getChartVersionValues))
-	apiv1.Methods("GET").Path("/assets/{repo}/{chartName}/versions/{version}/values.schema.json").Handler(WithParams(getChartVersionSchema))
+	// TODO: Check if we actually need multiple handlers here or can just use one path per endpoint.
```
[the remark this answers is not in the document]
Comment (absoludity) [reply cmd/assetsvc/main.go]: > I see changes in all these lines but I am actually not seeing any change thinking I guess you are just changing the call to listCharts to add WithParams but I am not sure.

The new versions are prefixed with `/ns/{namespace}` in the paths.

> 
> BTW potentially answering your TODO, it's necessary to repeat the Handler every time there is a new Query param

Why is that? I mean, if we have one path here which defines all possible query params, and one handler which handles that (including if those params are empty)?

File: cmd/assetsvc/mongodb_utils.go [hunk 2]
```diff
@@ -34,15 +34,22 @@ func newMongoDBManager(config datastore.Config) assetManager {
...
 
 	c := db.C(chartCollection)
 	pipeline := []bson.M{}
+	matcher := bson.M{}
+	if namespace != dbutils.AllNamespaces {
+		matcher["repo.namespace"] = namespace
+	}
 	if repo != "" {
-		pipeline = append(pipeline, bson.M{"$match": bson.M{"repo.name": repo}})
+		matcher["repo.name"] = repo
```
[the remark this answers is not in the document]
Comment (absoludity) [reply cmd/assetsvc/mongodb_utils.go]: Nope, that was a mistake. I meant to include it in the created pipeline. Updated, thanks.

File: cmd/assetsvc/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andresmgot) [reply cmd/assetsvc/main.go]: > Why is that? I mean, if we have one path here which defines all possible query params, and one handler which handles that (including if those params are empty)?

The matches were tricky. I am not sure if a request will match if there is a query parameter missing. In any case, I think that defining the query params in the path is not required so if you only have one backend function you can just try to read those params in the function.

**Comment by andresmgot** [comment]
Let me know if you need me to re-review this

**Comment by absoludity** [comment]
> Let me know if you need me to re-review this

Thanks, no, I'm just merging the other branches that you approve into this one which I'll then land.

**Pull request merged**